app/uk/gov/hmrc/cgtpropertydisposalsfrontend/repos/Repo.scala

```diff
@@ -60,25 +59,9 @@ trait Repo {
         .recover { case e ⇒ Left(Error(e)) }
     }
 
-  protected def store[A : Writes](id: String, a: A)(implicit
-    ec: ExecutionContext
-  ): Future[Either[Error, Unit]] =
-    preservingMdc {
-      cacheRepository
-        .createOrUpdate(Id(id), sessionKey, Json.toJson(a))
-        .map[Either[Error, Unit]] { dbUpdate ⇒
-          if (dbUpdate.writeResult.inError)
-            Left(
-              Error(
-                dbUpdate.writeResult.errmsg.getOrElse(
-                  "unknown error during inserting session data in mongo"
-                )
-              )
-            )
-          else
-            Right(())
-        }
-        .recover { case e ⇒ Left(Error(e)) }
-    }
-
+  protected def store[A : Writes](id: String, a: A)(implicit ec: ExecutionContext): Future[Either[Error, Unit]] =
+    cacheRepository
+      .put(id)(DataKey(sessionKey), a)
+      .map(_ => Right(()))
+      .recover { case e => Left(Error(s"Error in insertion $e")) }
```

Comment:
I think it should be s"Error in insertion $e.message" instead of s"Error in insertion $e", exception might contain some PI information

Reply:
Thank you changed it to a generic error message